core/app/models/spree/payment_method.rb

```diff
@@ -33,10 +39,23 @@ def payment_source_class
     end
 
     def self.available(display_on = 'both', store: nil)
-      all.select do |p|
-        p.active &&
-          (p.display_on == display_on.to_s || p.display_on.blank?) &&
-          (store.nil? || store.payment_methods.empty? || store.payment_methods.include?(p))
+      Spree::Deprecation.warn "Spree::PaymentMethod.available is deprecated."\
+        "Please use .active, .available_to_users, and .available_to_admin scopes instead."\
+        "For payment methods associated with a specific store, use your_store.payment_methods."
+
+      display_on = display_on.to_s
+
+      available_payment_methods =
+        case display_on
+        when 'front_end'
+          active.available_to_users
+        when 'back_end'
+          active.available_to_admin
+        else
+          active.available_to_users.available_to_admin + active.nil_available_to_users.nil_available_to_admin
```

Comment:
In the existing `core/spec/models/spree/payment_method_spec.rb` file's `describe .available` specs, there are four payment methods created, with `display_on` values of `'front_end'`, `'back_end'`, `'both'`, and `nil`.
The existing functionality filters with `(p.display_on == display_on.to_s || p.display_on.blank?)`. This behaviour of `.blank?` makes the live above (and the three value logic) necessary; the existing specs assert that there are two methods available for both. This should mean that there would be three each available for frontend and backend, as there are two for both and one each for those particular locations, however the specs show that this is not the case.